master/internal/stream/socketlike.go

```diff
@@ -0,0 +1,29 @@
+package stream
+
+import (
+	"fmt"
+
+	"github.com/gorilla/websocket"
+)
+
+type WebsocketLike interface {
+	ReadJSON(interface{}) error
+	Write(interface{}) error
+	Close() error
```

Comment:
can you defer a close right at the top of the Webscocket handler function, and leave this out of the interface?